### PR TITLE
Delete redundant examples from user gallery that are also present in the annotations tutorial

### PR DESCRIPTION
**Pull request by erg0dic**
## PR Summary
Tackles issue #24723. @rcomer @anntzer, which builds on the changed made by @story645 in the annotations' tutorial in the PR [#23606](https://github.com/matplotlib/matplotlib/pull/23606). 

I noticed there were about 7 obvious examples in the user gallery that could easily be removed so I did just that and added a
```
# .. redirect-from:: /gallery/userdemo/<gallery_example>
```
in the corresponding locations where the examples were used in the text of the `annotations.py` file. 

I'm not sure about the format for redirects so please let me know if they need to be in the docstrings in the header of `annotations.py`. 

## PR Checklist

**Documentation and Tests**
- [x] Has pytest style unit tests (and `pytest` passes)

### CODE REVIEW

File: tutorials/text/annotations.py [hunk 1]
```diff
@@ -388,6 +391,7 @@ def custom_box_style(x0, y0, width, height, mutation_size):
             arrowprops=dict(arrowstyle="simple",
                             connectionstyle="arc3,rad=-0.2"))
 
+# .. redirect-from:: /gallery/userdemo/annotate_simple03
```
Comment (jklymak) [inline tutorials/text/annotations.py:394]: I think these all need to be at the top - only /gallery/userdemo/annotate_simple01 is working as written.  (See https://output.circle-artifacts.com/output/job/9942d81b-1653-44c6-bc69-93f51ab0a293/artifacts/0/doc/build/html//gallery/userdemo/annotate_simple03.html versus 
https://output.circle-artifacts.com/output/job/9942d81b-1653-44c6-bc69-93f51ab0a293/artifacts/0/doc/build/html//gallery/userdemo/annotate_simple01.html)

File: tutorials/text/annotations.py [hunk 2]
```diff
@@ -295,7 +295,8 @@ def custom_box_style(x0, y0, width, height, mutation_size):
 #
 # Customizing annotation arrows
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-#
+# .. redirect-from:: /gallery/userdemo/annotate_simple01
+#
```
Comment (jklymak) [inline tutorials/text/annotations.py:299]: These flake8 errors need to be fixed...

**Review by jklymak — COMMENTED** [review]
Looks great, but I think redirects all need to be at the top, perhaps even before the title?

File: tutorials/text/annotations.py (hunk 2 above)
Comment (erg0dic) [reply tutorials/text/annotations.py]: > Looks great, but I think redirects all need to be at the top, perhaps even before the title?

Okay, done.

File: tutorials/text/annotations.py [hunk 3]
```diff
@@ -387,7 +393,6 @@ def custom_box_style(x0, y0, width, height, mutation_size):
             size=20, va="center", ha="center",
             arrowprops=dict(arrowstyle="simple",
                             connectionstyle="arc3,rad=-0.2"))
-
```
Comment (jklymak) [inline tutorials/text/annotations.py:390]: I don't think you can drop this line

File: tutorials/text/annotations.py [hunk 4]
```diff
@@ -635,7 +639,6 @@ def custom_box_style(x0, y0, width, height, mutation_size):
 ax.annotate("Test", xy=(0.5, 1), xycoords=("data", "axes fraction"))
 ax.axvline(x=.5, color='lightgray')
 ax.set(xlim=(0, 2), ylim=(1, 2))
-
```
Comment (jklymak) [inline tutorials/text/annotations.py:638]: Or this one.  Overall probably safer to not change any of these lines.

**Comment by erg0dic** [comment]
This is probably very late. But is there a way that I can render a local version of the website with the tutorials by any chance?

**Comment by jklymak** [comment]
https://matplotlib.org/devdocs/devel/documenting_mpl.html#building-the-docs. Note that if you want to just build certain directories, you can run `make html-skip-subdirs` and add or remove subdirs from `.mpl_skip_subdirs.yaml`.  This speeds up building the docs from many minutes to a few seconds.

**Comment by erg0dic** [comment]
Okay I think the redirects should be working now.

**Review by jklymak — APPROVED** [review]
We can squash merge, but if you want some git practice would you like to squash yourself?

**Comment by jklymak** [comment]
Ps if you try a squash for the first time maybe back up your branch, or at least do a git log before pushing to your fork.

**Comment by erg0dic** [comment]
Okay, I'll give it a go

**Comment by erg0dic** [comment]
I've squashed the commits and nothing too weird is happening from my end I think considering the commit changes. I'm not sure why some of the tests fail (the build docs' test in particular).

File: tutorials/text/annotations.py [hunk 5]
```diff
@@ -1,4 +1,11 @@
 r"""
+.. redirect-from:: /gallery/userdemo/annotate_simple01
+.. redirect-from:: /gallery/userdemo/annotate_simple02
+.. redirect-from:: /gallery/userdemo/annotate_simple03
+.. redirect-from:: /gallery/userdemo/annotate_simple04
+.. redirect-from:: /gallery/userdemo/anchored_box04
+.. redirect-from:: /gallery/userdemo/annotate_simple_coord01
+.. redirect-from:: /gallery/userdemo/annotate_simple_coord03
```
Comment (rcomer) [inline tutorials/text/annotations.py:8]: The doc build is failing with this (fairly cryptic) warning

`/home/circleci/project/doc/tutorials/text/annotations.rst:31: WARNING: Explicit markup ends without a blank line; unexpected unindent.`

If you look at line 31 of your locally built `doc/tutorials/text/annotations.rst`, it corresponds to this point in the python file.  So I _think_ you just need to add a blank line between the "redirect-from"s and the title.

Comment (erg0dic) [reply tutorials/text/annotations.py:8]: Thanks for the explanation! I've added a blank line.

**Comment by erg0dic** [comment]
Okay, I think the docs build now. I don't think the `codecov` error is relevant. This can probably merge now.

**Comment by jklymak** [comment]
I squashed this @erg0dic 

Thanks for the contribution, and we hope to hear from you again.

**Pull request merged**